metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/LinkedProcessor.java

```diff
@@ -0,0 +1,64 @@
+package eu.europeana.metis.mediaprocessing.extraction;
+
+import eu.europeana.metis.mediaprocessing.exception.MediaExtractionException;
+import eu.europeana.metis.mediaprocessing.model.Resource;
+import eu.europeana.metis.mediaprocessing.model.ResourceExtractionResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LinkedProcessor implements MediaProcessor {
```

Comment:
The idea behind of this linked processor is using the `chain of responsability` pattern. 
So when a media processing is requested will go first checking if it is an ombed object, if not then will go on the next on the chain and test with a text processor.
This gives a starting point to the ombed and the text processor. because both go on the same media types (application/json and application/xml) it could be an ombed or just normal xml or normal json.

Reply:
Ok let's assume the chaining here is applied - 
You are just setting:
`this.linkedProcessor.setNextProcessor(oEmbedProcessor);`
And all calls inside delegate to oEmbedProcessor calls.
And then in the chooseMediaProcessor method you have:
`case TEXT, OTHER -> processor = linkedProcessor;`
Why not just set:
`case TEXT, OTHER -> processor = oEmbedProcessor;`?
The "LinkedProcessor" is in fact redundant isn't it?

Reply:
Yes. I did it on purpose. To remind when reading the code. if someone else reads it they will come to notice the chain of processors. I opt for verbosity in this situation, and later I thought it will be easy to catch up.

Reply:
Then I would definitely suggest removing this.
Think about it, you've added a whole class just for a reminder and that does not add anything else to the code base..
This should be javadoc and if need be with a more extended explanation.

Reply:
The challenge I found is that putting in front of the chain the oEmbedProcessor felt awkward, because it will obscure the TextProcessor. And Having the linked processor will give the idea that connected things were happening inside.